com.unity.sg2/Editor/GraphUI/DataModel/Constants/SamplerTypeConstant.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using UnityEditor.ShaderGraph.GraphDelta;
+using UnityEngine;
+using UnityEngine.GraphToolsFoundation.Overdrive;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    internal struct SamplerStateData
+    {
+        public SamplerStateType.Filter filter;
+        public SamplerStateType.Wrap wrap;
+        public bool depthCompare;
+        public SamplerStateType.Aniso aniso;
+    }
+
+    public class SamplerStateTypeConstant : BaseShaderGraphConstant
+    {
+        override protected object GetValue()
+        {
+            return new SamplerStateData {
+            filter = SamplerStateType.GetFilter(GetField()),
+            wrap = SamplerStateType.GetWrap(GetField()),
+            depthCompare = SamplerStateType.GetDepthComparison(GetField()),
+            aniso = SamplerStateType.GetAniso(GetField()) };
+        }
+
+        override protected void SetValue(object value)
+        {
+            var smp = (SamplerStateData)value;
+            SamplerStateType.SetFilter(GetField(), smp.filter);
+            SamplerStateType.SetWrap(GetField(), smp.wrap);
+            SamplerStateType.SetDepthComparison(GetField(), smp.depthCompare);
+            SamplerStateType.SetAniso(GetField(), smp.aniso);
+    }
+
+        override public object DefaultValue => Activator.CreateInstance(Type);
+        override public Type Type => typeof(SamplerStateData);
+        override public TypeHandle GetTypeHandle() => ShaderGraphExampleTypes.SamplerStateTypeHandle;
+    }
+
```

Comment:
Don't love this- but SamplerStates don't have a public unity standard data object for handling, representing, or manipulating them currently. There's a couple different ways we could setup a constant for them, but this is probably fine and simpler for UI impl.